### PR TITLE
Retry ES Calls using httpx

### PR DESCRIPTION
**Pull request by tybritten**

### CODE REVIEW

File: indexer.go [hunk 1]
```diff
@@ -132,7 +162,10 @@ func CleanupIndexes(url string, alias string) error {
 func MakeJSONRequest(method string, url string, body string, jsonStruct interface{}) (*http.Response, error) {
 	req, _ := http.NewRequest(method, url, bytes.NewReader([]byte(body)))
 	req.Header.Add("Content-Type", "application/json")
-	resp, err := http.DefaultClient.Do(req)
+	retrycount := 5
+	initialBackoff := 2 * time.Second
+	retryConfig := ElasticRetries(initialBackoff, retrycount)
```
Comment (rowanseymour) [inline indexer.go]: this might as well be a global save creating it for every request

File: go.sum [hunk 2]
```diff
@@ -18,23 +38,78 @@ github.com/naoina/go-stringutil v0.1.0 h1:rCUeRUHjBjGTSHl0VC00jUPLz8/F9dDzYI70Hz
...
+gopkg.in/check.v1 v0.0.0-20161208181325-20d25e280405 h1:yhCVgyC4o1eVCa2tZl7eS0r+SDo693bJlVdllGtEeKM=
+gopkg.in/check.v1 v0.0.0-20161208181325-20d25e280405/go.mod h1:Co6ibVJAznAaIkqp8huTwlJQCZ016jof/cbN4VW5Yz0=
+gopkg.in/fsnotify.v1 v1.4.7/go.mod h1:Tz8NjZHkW78fSQdbUxIjBTcgA1z1m8ZHf0WmKUhAMys=
 gopkg.in/gemnasium/logrus-airbrake-hook.v2 v2.1.2 h1:OAj3g0cR6Dx/R07QgQe8wkA9RNjB2u4i700xBkIT4e0=
 gopkg.in/gemnasium/logrus-airbrake-hook.v2 v2.1.2/go.mod h1:Xk6kEKp8OKb+X14hQBKWaSkCsqBpgog8nAV2xsGOxlo=
+gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7 h1:uRGJdciOHaEIrze2W8Q3AKkepLTh2hOroT7a+7czfdQ=
+gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7/go.mod h1:dt/ZhP58zS4L8KSrWDmTeBkI65Dw0HsyUHuEVlX15mw=
+gopkg.in/yaml.v2 v2.2.4/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
+gopkg.in/yaml.v2 v2.3.0 h1:clyUAQHOM3G0M3f5vQj7LuJrETvjVot3Z5el9nffUtU=
+gopkg.in/yaml.v2 v2.3.0/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
```
Comment (rowanseymour) [inline go.sum:105]: maybe run `go mod tidy` if you didn't to make sure only things we use are in here

**Review by rowanseymour — CHANGES_REQUESTED** [review]

File: indexer.go [hunk 3]
```diff
@@ -11,11 +11,46 @@ import (
...
+	}
+
+	response.Body = ioutil.NopCloser(bytes.NewBuffer(bodyBytes))
+	return false
+}
+
+// retry settings
+var retrycount = 5
+var initialBackoff = 1 * time.Second
+var retryConfig = ElasticRetries(initialBackoff, retrycount)
```
Comment (rowanseymour) [inline indexer.go]: this isn't actually being used.. also variables above can be inline

**Review by rowanseymour — CHANGES_REQUESTED** [review]

File: indexer.go [hunk 4]
```diff
@@ -132,7 +165,7 @@ func CleanupIndexes(url string, alias string) error {
 func MakeJSONRequest(method string, url string, body string, jsonStruct interface{}) (*http.Response, error) {
 	req, _ := http.NewRequest(method, url, bytes.NewReader([]byte(body)))
 	req.Header.Add("Content-Type", "application/json")
-	resp, err := http.DefaultClient.Do(req)
+	resp, err := httpx.Do(http.DefaultClient, req, ElasticRetries(), nil)
```
Comment (rowanseymour) [inline indexer.go]: but now we're back to creating the retry config for every request 😄 

my thinking was one global var called `retryConfig`, initialize that in `init()` and use it for all requests

**Comment by tybritten** [comment]
haha whoops, ok now I know what you meant.

**Review by rowanseymour — APPROVED** [review]

**Comment by rowanseymour** [comment]
@tybritten I've merged this and it's part of 5.7.1 but so is https://github.com/nyaruka/rp-indexer/pull/29 which you can only run if your RapidPro instance is up to date with our 5.7 branch... we're making a lot of changes this week

**Pull request merged**